x-pack/legacy/plugins/actions/server/builtin_action_types/email.ts

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { schema, TypeOf } from '@kbn/config-schema';
+
+import { sendEmail } from './lib/send_email';
+import { ActionType, ActionTypeExecutorOptions } from '../types';
+
+const PORT_MAX = 256 * 256 - 1;
+
+export type ActionTypeConfigType = TypeOf<typeof ActionTypeConfig.schema>;
+
+const ActionTypeConfig = {
+  path: 'actionTypeConfig',
+  schema: schema.object({
+    service: schema.maybe(schema.string()),
+    host: schema.maybe(schema.string()),
+    port: schema.maybe(schema.number({ min: 1, max: PORT_MAX })),
+    secure: schema.maybe(schema.boolean()),
+    user: schema.string(),
+    password: schema.string(),
+    from: schema.string(),
+  }),
+};
+
+const unencryptedConfigProperties = ['service', 'host', 'port', 'secure', 'user', 'from'];
```

Comment:
WORKSFORME.

As an aside, I think this particular example shows that it's probably going to be easier to list the encrypted config properties, rather than the unencrypted ones, because there will likely be more unencrypted values than encrypted ones.  But ... I want to re-look at how to specify the encrypted things differently anyway, and what we have now seems migratable to something different for a > 7.3 releases, so ... deferring :-)